src/machine/machine_gameboyadvance.go

```diff
@@ -18,27 +120,128 @@ func (p Pin) Set(value bool) {
 	// do nothing
 }
 
-var Display = FramebufDisplay{(*[160][240]volatile.Register16)(unsafe.Pointer(uintptr(0x06000000)))}
-
-type FramebufDisplay struct {
-	port *[160][240]volatile.Register16
+// Displays is a convenience container for the various mode-based displays.
+type Displays struct {
+	Mode3 DisplayMode3
```

Comment:
Why do you embed the `DisplayMode3` struct inside the `Displays` struct? Commits later in the chain don't add any extra fields to this struct so I don't really see the point.

In particular, I think it would be interesting if the one display mode that has a standard interface (meaning, it is pixel based), could be used in a portable way. So if there is a board with a similar display, it could be used the exact same way (similar to how src/examples/blinky1 is portable to most boards without changes). I don't see that happening with `machine.Display.Mode3`.

Reply:
I don't have a pull request for it, but modes 3, 4, and 5 are all bitmapped modes, and could all implement Displayer.

I guess what you're suggesting is that instead of saying `machine.Display.Mode3.Configure()` you could make it `machine.Display.Configure(machine.Mode3)` and have its methods parameterized on the configured mode?

Reply:
As an example of another field of the Displays struct:

[display.Mode0.Configure](https://github.com/kylelemons/tinygo/blob/gameboy-advance/src/machine/machine_gameboyadvance.go#L248) is the way that I have set it up to enable the basic tilemapped mode.

Reply:
I quite prefer `machine.Display.Configure(machine.Mode3)`, myself.

Reply:
My only argument against that is that the different modes might require separate sets of arguments to configure them.  I recognize that it is convenient to make demos work across platforms, but the GBA display is an entirely different animal from an LED or LCD matrix -- it's basically a GPU.

But, I think I can probably still work within the constraint of a `machine.Display.Configure()` API, though it makes it slightly more indirect.  I could do something like

```
type DisplayRegs ...
func (d *DisplayRegs) Configure(options ...DisplayOption)
type DisplayOption func(*DisplayRegs)
func WithMode3() DisplayOption
func WithMode5(activePage int) DisplayOption
func WithMode0(activeBackground int, bitdepth Depth) DisplayOption
```

where the "default" is Mode3, though that feels like the kind of higher-level API that I would put in /x/drivers rather than machine.  WDYT?

Reply:
We do not really want a dependency from this repo to the `drivers` repo, so we already have a bit of a question of how to best implement this. 

I think the style of initialization that you are proposing seems like the cleanest. This would be the end result, right?

```
machine.Display.Configure(machine.WithMode5(16))
```